cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java

```diff
@@ -139,8 +141,24 @@ public static void pushPackages(String packageName, String sourceRoot, String in
         }
         
         if (installToRepo == null) {
-            // Get access token
-            String accessToken = checkAccessToken();
+            // The access token can be specified as an environment variable or in 'Settings.toml'. First we would
+            // check if the access token was specified as an environment variable. If not we would read it from
+            // 'Settings.toml'.
+
+            // Check if the access token is specified as an environment variable
+            String accessToken = System.getenv("BALLERINA_CENTRAL_ACCESS_TOKEN");
+            if (accessToken != null) {
```

Comment:
I think we can cleanup this code design a bit. Rather than, doing this logic here, and creating redundant code, update the code in "TomlParserUtils.readSettings()" method. Here, simply check if a Settings.toml is there in the home directory, if not, create empty values for those, Central object etc... and also check for the existence of the environment variable here for "accessToken", or else, keep it the original value (coming from Settings.toml, if not there, empty). Then, the rest of the existing logic can be kept in-tact, and no code duplication is required.

Reply:
Refactored the code with https://github.com/ballerina-platform/ballerina-lang/pull/14242/commits/051de5130e23039496be15fe8469f93bb3505e10